homeassistant/components/lektrico/sensor.py

```diff
@@ -0,0 +1,206 @@
+        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+        value_fn=lambda x: x.current,
+    ),
+    LektricoSensorEntityDescription(
+        key="instant_power",
+        name="Instant Power",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.POWER,
+        native_unit_of_measurement=POWER_KILO_WATT,
+        value_fn=lambda x: x.instant_power,
+    ),
+    LektricoSensorEntityDescription(
+        key="session_energy",
+        name="Session Energy",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        value_fn=lambda x: x.session_energy,
+    ),
+    LektricoSensorEntityDescription(
+        key="temperature",
+        name="Temperature",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        value_fn=lambda x: x.temperature,
+    ),
+    LektricoSensorEntityDescription(
+        key="total_charged_energy",
+        name="Total Charged Energy",
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        value_fn=lambda x: x.total_charged_energy,
+    ),
+    LektricoSensorEntityDescription(
+        key="voltage",
+        name="Voltage",
+        device_class=SensorDeviceClass.VOLTAGE,
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_VOLT,
+        value_fn=lambda x: x.voltage,
+    ),
+    LektricoSensorEntityDescription(
+        key="install_current",
+        name="Install Current",
+        device_class=SensorDeviceClass.CURRENT,
+        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+        value_fn=lambda x: x.install_current,
+    ),
+    LektricoSensorEntityDescription(
+        key="dynamic_current",
+        name="Dynamic Current",
+        device_class=SensorDeviceClass.CURRENT,
+        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+        value_fn=lambda x: x.dynamic_current,
+    ),
+    LektricoSensorEntityDescription(
+        key="led_max_brightness",
+        name="Led Brightness",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.ILLUMINANCE,
+        native_unit_of_measurement=PERCENTAGE,
+        value_fn=lambda x: x.led_max_brightness,
+    ),
+    LektricoSensorEntityDescription(
+        key="headless",
+        name="No Authentication",
+        value_fn=lambda x: x.headless,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Lektrico charger based on a config entry."""
+    _lektrico_device: LektricoDevice = hass.data[DOMAIN][entry.entry_id]
+
+    sensors = [
+        LektricoSensor(
+            sensor_desc,
+            _lektrico_device.serial_number,
+            _lektrico_device.board_revision,
```

Comment:
Since `serial_number` and `board_revision` are always a part of `_lektrico_device`, don't pass them to `__init__`. Instead in `__init__` just grab those values out of `_lektrico_device` in `__init__`.

Reply:
I removed them, thank you.